alphabet-soup/src/test/scala/io/typechecked/alphabetsoup/AtomMacroSpec.scala

```diff
@@ -152,6 +152,35 @@ class AtomMacroSpec extends FlatSpec with Matchers {
    illTyped("@Atomic object Foo",".*Invalid: Can not annotate structure with @Atomic.*")
 
   }
+  it should "create an implicit Atom for a class with a type parameter" in {
+    @Atomic class Foo[A]
+    implicitly[Atom[Foo[String]]]
+  }
+  it should "create an implicit Atom for a class with a higher-kinded type parameter" in {
+    @Atomic class Foo[A[_]]
+    implicitly[Atom[Foo[Option]]]
+  }
+  it should "create an implicit Atom for a class with many type parameters" in {
+    @Atomic class Foo[A[_], B, C]
+    implicitly[Atom[Foo[Option, String, (Boolean, List[String])]]]
+  }
+  it should "create an implicit Atom for a class with a type parameter with a type bound" in {
```

Comment:
Could you add a test demonstrating it works for the following case:

```scala
class Foo[T[_]: Functor]
```

Reply:
this isn't really quite the same thing as the type bound tests, as that would get desugared to `class Foo[T[_]](implicit ev: Functor[T])`. You would still have a Atom for any old `T[_]`